core/src/main/java/io/grpc/internal/GzipInflatingBuffer.java

```diff
@@ -0,0 +1,514 @@
+
+  private boolean processHeaderCompressionMethod() throws ZipException {
+    if (gzipMetadataReader.readableBytes() > 0) {
+      if (gzipMetadataReader.readUnsignedByte() != 8) {
+        throw new ZipException("Unsupported compression method");
+      }
+      state = State.HEADER_FLAG;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  private boolean processHeaderFlag() {
+    int requiredBytes =
+        GZIP_HEADER_MIN_SIZE - UNSIGNED_SHORT_SIZE /* header magic */ - 1 /* compression method */;
+    if (gzipMetadataReader.readableBytes() >= requiredBytes) {
+      gzipHeaderFlag = gzipMetadataReader.readUnsignedByte();
+      gzipMetadataReader.skipBytes(6 /* remaining header bytes */);
+      state = State.HEADER_EXTRA_LEN;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  private boolean processHeaderExtraLen() {
+    if ((gzipHeaderFlag & HEADER_EXTRA_FLAG) != HEADER_EXTRA_FLAG) {
+      state = State.HEADER_NAME;
+      return true;
+    } else if (gzipMetadataReader.readableBytes() >= UNSIGNED_SHORT_SIZE) {
+      headerExtraToRead = readUnsignedShort(gzipMetadataReader);
+      state = State.HEADER_EXTRA;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  private boolean processHeaderExtra() {
+    if (gzipMetadataReader.readableBytes() >= headerExtraToRead) {
+      gzipMetadataReader.skipBytes(headerExtraToRead);
+      state = State.HEADER_NAME;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  private boolean processHeaderName() {
+    if ((gzipHeaderFlag & HEADER_NAME_FLAG) != HEADER_NAME_FLAG) {
+      state = State.HEADER_COMMENT;
+      return true;
+    } else if (readBytesUntilZero()) {
+      state = State.HEADER_COMMENT;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  private boolean processHeaderComment() {
+    if ((gzipHeaderFlag & HEADER_COMMENT_FLAG) != HEADER_COMMENT_FLAG) {
+      state = State.HEADER_CRC;
+      return true;
+    } else if (readBytesUntilZero()) {
+      state = State.HEADER_CRC;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  private boolean processHeaderCrc() throws ZipException {
+    if ((gzipHeaderFlag & HEADER_CRC_FLAG) != HEADER_CRC_FLAG) {
+      state = State.INITIALIZE_INFLATER;
+      return true;
+    } else if (gzipMetadataReader.readableBytes() >= UNSIGNED_SHORT_SIZE) {
+      int desiredCrc16 = (int) crc.getValue() & 0xffff;
+      if (desiredCrc16 != readUnsignedShort(gzipMetadataReader)) {
+        throw new ZipException("Corrupt GZIP header");
+      }
+      state = State.INITIALIZE_INFLATER;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  private boolean processTrailer() throws ZipException {
+    if (gzipMetadataReader.readableBytes() >= GZIP_TRAILER_SIZE) {
+      if (crc.getValue() != (readUnsignedInt(gzipMetadataReader))
+          ||
+          (readUnsignedInt(gzipMetadataReader) != expectedGzipTrailerIsize)) {
```

Comment:
Put `readUnsignedInt` on the same side of the comparison? It's on the left here, but in the right above.

Reply:
Done